### ISSUE TITLE
error message

### ISSUE THREAD

**Issue by Lostmp5** [zephyrj/sim-racing-tools#10]
![image](https://user-images.githubusercontent.com/83926241/187666525-3e8e5674-09a1-40ce-aeb8-3d6456777184.png)

**Comment by zephyrj** [comment]
Looks like you need to exit and then re-enter the virtual -enviroment so that the ac-tools executable can be found on the PATH

**Comment by zephyrj** [comment]
Inside that folder you can type `deactivate` to exit the virtual-enviroment and then, inside the same folder, type `.\Scripts\activate.bat` to enter the virtual environment again

**Comment by Lostmp5** [comment]
i dont have much knowlege with python and programing so sorry if i say something wrong. I think the package doesnt install in the virtual enviroment but on my appdata.

**Comment by zephyrj** [comment]
A python virtual-environment should be self-contained so it shouldn't mess around with anything in appdata if it is working correctly. Assuming the virtual environment has been setup correctly and you have installed sim-racing tools inside of it then it should only be installed and accessible from inside the virtual environment itself.

Here is the output on my machine:

## Outside virtual environment
```commandline
C:\Users\user\sim-racing-tool-venv>ac-tools
'ac-tools' is not recognized as an internal or external command,
operable program or batch file.
```

## Inside virtual environment
```commandline
C:\Users\user\sim-racing-tool-venv>.\Scripts\activate.bat

(sim-racing-tool-venv) C:\Users\josykes\sim-racing-tool-venv>ac-tools --help
usage: ac-tools.exe [-h] {clone-car,swap-automation-engine} ...

Tools for working with Assetto Corsa (AC) content

options:
  -h, --help            show this help message and exit

Commands:
  {clone-car,swap-automation-engine}
    clone-car           Clone a car from the AC cars directory. Note: You must have already extracted the data.acd
                        file to a data directory
    swap-automation-engine
                        Takes an engine from an automation car exported to BeamNG, creates the AC engine parameter
                        data and puts that data into the provided car folder

```
----------

Taking a closer look at your output in your first message it looks like you are using Windows Powershell rather than the traditional command prompt - is this the case?

I can't say I know much about Powershell or whether or not the install process will work on it the same way - I would suggest using command prompt instead if you are able. If you are not able then I have noticed that there is a script called `.\Scripts\Activate.ps1` which looks like it can be used in Powershell (replacing the activate.bat script) - this seems to work for me too:
```Powershell
PS C:\Users\user> cd .\sim-racing-tool-venv\
PS C:\Users\user\sim-racing-tool-venv> .\Scripts\Activate.ps1
(sim-racing-tool-venv) PS C:\Users\josykes\sim-racing-tool-venv> ac-tools --help
usage: ac-tools.exe [-h] {clone-car,swap-automation-engine} ...
```
-----
If you have been using Powershell for the whole install process then I suspect there is a chance that things have been installed outside of the virtual-environment, or just not installed correctly at all.

If you are still having issues I would suggest deleting the existing folder you created and going through the process of creating the virtual-environment and installing sim-racing-tools again. I would suggest doing this using command prompt as that is what I have tested using. 

If for some reason you have to use Powershell then I would still suggest starting again and following the install steps but whenever you see:
```
.\Scripts\activate.bat
```
replace this with:
```
.\Scripts\Activate.ps1
```

Hope this helps :)

**Comment by zephyrj** [comment]
It has suddenly occurred to me that you may be using Windows 11 (which I don't have access to) which may be defaulting to using PowerShell.

This link may help explain how to use command prompt instead:
https://www.minitool.com/news/open-command-prompt-windows-11.html

**Comment by Lostmp5** [comment]
So the problem was windows powershell is the default terminal in windows 11. I now switched to cmd and i get the virtual environment to work. I will now try to swap an engine.
![image](https://user-images.githubusercontent.com/83926241/188149962-4c5997f0-b585-4051-b40d-9422b54ce4ec.png)

**Comment by Lostmp5** [comment]
I now get this error
![image](https://user-images.githubusercontent.com/83926241/188152507-0abfe2d5-2c09-43d3-b79d-02894b417d7e.png)

**Comment by zephyrj** [comment]
Looks like it might be failing to find the ac installation. Do you know the path of your AssettoCorsa install? I wonder if Windows 11 puts the Steam games into a different path to what it's looking for. 

On my Windows 10 install the install is in `C:\Program Files (x86)\Steam\steamapps\common\assettocorsa`

**Comment by zephyrj** [comment]
A another point that has come to mind is that perhaps there is a anti-virus program blocking access to the Program Files folder. I use BitDefender and don't seem to have any issues but perhaps it's playing a role in this?

Another issue that you will encounter, hopefully once this issue is solved, is that you will need to move your engine the BeamNG mod folder (rather than `A:\test\2JZ`)

From https://github.com/zephyrj/sim-racing-tools/blob/main/sim_racing_tools/assetto_corsa/docs/swap_automation_engine.md:
> - the BeamNG mod that you wish to export the engine from resides within the standard BeamNG mod folder or in Documents/BeamNG.drive/mods - if you don't have BeamNG installed then you will need to create this folder.

**Comment by Lostmp5** [comment]
I have assetto corsa intsalled on A:\Programme\steam\steamapps\common\assettocorsa

**Comment by zephyrj** [comment]
Ah ok - so I guess you have your steam library on a different Disk to the system install? I'll look at updating it so it can account for this

**Comment by Lostmp5** [comment]
thanks

**Comment by Lostmp5** [comment]
Thank you very much, i will now try the new version but i wanted to inform you that the downloadable 7zip folder on your website gets blocked by windows antivirus.

**Comment by Lostmp5** [comment]
It Worked! But i had to open the exe in Administrator mode, and big up for the user interface.

**Comment by zephyrj** [comment]
Excellent, I'm glad it worked for you

**Comment by zephyrj** [comment]
I don't suppose you could tell me what Windows Anti-Virus reported the file as? I think I can send the file to Microsoft for them to verify that it isn't malware. I think it's listed inside the  Virus & threat protection -> Protection history section of the app

**Comment by Lostmp5** [comment]
And can you maybe implement the option to also transfer the sound of the beamng mod?

**Comment by zephyrj** [comment]
Thanks. I have plans to add engine sounds but I don't actually know how to make that work yet. I will be actively looking into
